auth/src/VSCodeAzureSubscriptionProvider.ts

```diff
@@ -58,8 +58,8 @@ export class VSCodeAzureSubscriptionProvider extends vscode.Disposable implement
      *
      * @returns A list of tenants.
      */
-    public async getTenants(): Promise<TenantIdDescription[]> {
-        const { client } = await this.getSubscriptionClient();
+    public async getTenants(session?: vscode.AuthenticationSession): Promise<TenantIdDescription[]> {

```

Comment:
I think we should change the parameter to `account?: vscode.AuthenticationSessionAccountInformation` and then acquire the session within the function. That way consumers only have to list accounts and then provide them instead of listing accounts, getting the session, then passing that in.

See what I mean in the RGs PR here: https://github.com/microsoft/vscode-azureresourcegroups/pull/869/files#diff-c145c307da953aefde25e0a2d004387be9aba78ad9e596af5628ef25b4074246R57-R60


```ts
                const accounts = await vscode.authentication.getAccounts(getConfiguredAuthProviderId());
                for (const account of accounts) {
                    const session = await vscode.authentication.getSession(getConfiguredAuthProviderId(), getScopes(undefined), { account: account });
                    const tenants = await this.getTenants(session);
```

It eliminates the need to expose the `getScopes()` logic.

Reply:
I tried this out locally and it seemed to work well.

Reply:
Ok sounds good I will change it over to this logic.